src/main/java/com/paneedah/mwc/groovyscript/script/CraftingStation.java

```diff
@@ -0,0 +1,171 @@
+        removeinGroup(ingredient, CraftingGroup.valueOf(group));
+    }
+
+    public void addRecipe(IModernCrafting crafting) {
+        registerRecipe(crafting);
+        this.addScripted(crafting);
+    }
+
+    /**
+     * Start Recipe Builder for GroovyScript.
+     */
+    public RecipeBuilder recipeBuilder() {
+        return new RecipeBuilder();
+    }
+
+    public static class RecipeBuilder extends AbstractRecipeBuilder<GSCrafting> {
+
+        @Override
+        public String getErrorMsg() {
+            return "Error adding Modern Warefare Cubed Workbench Recipe: "+this.name;
+        }
+
+        @Override
+        public void validate(GroovyLog.Msg msg) {
+            //max input == container size of crafting station.
+            this.validateItems(msg, 1, 27,1,1);
+        }
+
+        private final ArrayList<Double> yields = new ArrayList<>();
+        private double yield = 1.0;
+
+        private CraftingGroup group = GUN;
+
+        /**
+         * Add Ingredient of recipe With ItemStack
+         *
+         * @param ingredient crafting ingredient
+         * @return AbstractRecipeBuilder
+         */
+        @Override
+        public AbstractRecipeBuilder<GSCrafting> input(IIngredient ingredient) {
+            this.yields.add(this.yield);
+            return super.input(ingredient);
+        }
+
+        /**
+         * Set return Yield of upcoming Ingredient.
+         *
+         * @param yield return Yield of upcoming ingredients
+         * @return this RecipeBuilder
+         */
+        public RecipeBuilder setYield(double yield) {
+            this.yield = yield;
+            return this;
+        }
+
+        /**
+         * Set Category of Recipe
+         *
+         * @param group: name of the Group. valid values = "GUN", "ATTACHMENT_NORMAL", "ATTACHMENT_MODIFICATION", "BULLET", "MAGAZINE"
+         * @return this RecipeBuilder
+         */
+        public RecipeBuilder setGroup(String group) {
+            this.group = CraftingGroup.valueOf(group);
+            return this;
+        }
+
+        /**
+         * Finalize the recipe and register it
+         *
+         * @return Finished Recipe
+         */
+        @Override
+        public GSCrafting register() {
+            if (!this.validate())
+                return null;
+
+            ArrayList<CraftingEntry> entries = new ArrayList<>();
+
+            int i = 0;
+
+            for (IIngredient ingredient : this.input) {
+                if (ingredient instanceof OreDictIngredient) {
+                    OreDictIngredient dict = (OreDictIngredient) ingredient;
+                    entries.add(new CraftingEntry(dict.getFirst().getItem(), dict.getOreDict(), dict.getAmount(), this.yields.get(i)));
+                } else
+                    entries.add(new CraftingEntry(ingredient.getMatchingStacks()[0].getItem(), ingredient.getAmount(), this.yields.get(i)));
+                i++;
+            }
```

Comment:
Switching to a non-enhanced for loop would be cleaner (since no need for external index management)